pkg/backend/httpstate/backend.go

```diff
@@ -762,38 +762,51 @@ func (b *cloudBackend) RemoveStack(ctx context.Context, stack backend.Stack, for
 	return b.client.DeleteStack(ctx, stackID, force)
 }
 
-func (b *cloudBackend) RenameStack(ctx context.Context, stack backend.Stack, newName tokens.QName) error {
+func (b *cloudBackend) RenameStack(ctx context.Context, stack backend.Stack,
+	newName tokens.QName) (backend.StackReference, error) {
 	stackID, err := b.getCloudStackIdentifier(stack.Ref())
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Support a qualified stack name, which would also rename the stack's project too.
 	// e.g. if you want to change the project name on the Pulumi Console to reflect a
 	// new value in Pulumi.yaml.
 	newRef, err := b.ParseStackReference(string(newName))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	newIdentity, err := b.getCloudStackIdentifier(newRef)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if stackID.Owner != newIdentity.Owner {
-		url := "."
+		errMsg := "New stack owner, %s, does not match existing owner, %s.\n\n"
+
+		qn, _ := b.parseStackName(string(newName))
+		if qn.Owner == "" {
+			errMsg += fmt.Sprintf(
+				"       Did you forget to include the owner name? If yes, rerun the command as follows:\n\n"+
+					"           $ pulumi stack rename %s/%s\n\n",
+				stackID.Owner, newName)
+		}
 
+		url := "."
 		if consoleURL, err := b.StackConsoleURL(stack.Ref()); err == nil {
-			url = ":\n" + consoleURL + "/settings/options"
+			url = ":\n\n           " + consoleURL + "/settings/options"
 		}
-		errMsg := "You cannot transfer stack ownership via a rename. If you wish to transfer ownership\n" +
-			"of a stack to another organization, you can do so in the Pulumi Console by going to the\n" +
-			"\"Settings\" page of the stack and then clicking the \"Transfer Stack\" button" + url
+		errMsg += "       You cannot transfer stack ownership via a rename. If you wish to transfer ownership\n" +
+			"       of a stack to another organization, you can do so in the Pulumi Console by going to the\n" +
+			"       \"Settings\" page of the stack and then clicking the \"Transfer Stack\" button" + url
 
-		return errors.Errorf(errMsg)
+		return nil, errors.Errorf(errMsg, stackID.Owner, newIdentity.Owner)
```

Comment:
I'd suggest we just use `errors.New` here, and use `errMsg := fmt.Spritnf(..., stackID.Owner, newIdentity.Owner)` earlier. That way we don't need to "remember" there are 2x values to be filled into the error string later.